src/library/VUMeter.js

```diff
@@ -0,0 +1,112 @@
+    /** @private @const {!canvas} Selected waveform canvas element */
+    this.canvas_ = document.querySelector(canvasId);
+    /** @private @const {!canvasContext} Canvas context */
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    /** @private @const {!width} Selected waveform canvas width */
+    this.width_ = this.canvas_.width;
+    /** @private @const {!height} Selected waveform canvas height */
+    this.height_ = this.canvas_.height;
+    /** @private @const {!minDisplayDecible_} 
+     *    Minimum decible of the VU meter. Any decibel lower than this value
+     *    is consider slient. Due to the calculation and graph convinience,
+     *    we convert it to positive. */
+    this.minDisplayDecible_ = Math.abs(minDecible);
+    /** @private @const {!fifo} First in first out queue for control
+     *    the VU meter updating speed. */
+    this.fifo_ = new FIFO(fifoSize);
+    /** @private @const {!analyser} AnalyserNode of audio context */
+    this.analyser_ = analyserNode;
+    this.analyser_.fftSize = fftSize;
+    /** @private @const {!bufferLength} Total number of data 
+     *    points avliable to AudioContext sampleRate */
+    this.bufferLength_ = this.analyser_.frequencyBinCount;
+    /** @private @const {!dataArray} AnalyserNode of audio context */
+    this.dataArray_ = new Float32Array(this.bufferLength_);
+  }
+
+  /**
+   * This is the draw function for VUMeter.
+   */
+  draw() {
+    this.analyser_.getFloatTimeDomainData(this.dataArray_);
+
+    const rootMeanSquare = this.calculateRMS(this.dataArray_);
+    const decibel = 20 * Math.log10(rootMeanSquare);
+    let absDecibel = Math.abs(decibel);
+    this.fifo_.add(absDecibel);
+    let minDecibel = this.fifo_.getMinValue();
+    let meterHeight = minDecibel > this.minDisplayDecible_ ?
+      this.height_ : minDecibel / this.minDisplayDecible_ * this.height_;
+
+    this.canvasContext_.clearRect(0, 0, this.width_, this.height_);
+
+    this.canvasContext_.fillStyle = '#000';
+    this.canvasContext_.fillRect(0, 0, this.width_, this.height_);
+    this.canvasContext_.fillStyle = 'green';
+    this.canvasContext_.fillRect(0, meterHeight, this.width_, this.height_ - meterHeight);
+
+    this.canvasContext_.strokeStyle = 'black';
+    this.canvasContext_.lineWidth = 1;
+    this.canvasContext_.globalAlpha = 0.3;
+    this.canvasContext_.beginPath();
+    this.canvasContext_.moveTo(0, this.height_ * 0.2);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.2);
+    this.canvasContext_.moveTo(0, this.height * 0.4);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.4);
+    this.canvasContext_.moveTo(0, this.height_ * 0.6);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.6);
+    this.canvasContext_.moveTo(0, this.height_ * 0.8);
+    this.canvasContext_.lineTo(this.width_, this.height_ * 0.8);
+    this.canvasContext_.stroke();
+  }
+
+  /**
+   * Calculation of the root mean square of input array.
```

Comment:
This is a method, so start with a verb. Perhaps:

"Calculate a root mean square value from `inputArray`".